src/job.ts

```diff
@@ -53,17 +88,49 @@ export class CronJob {
 	}
 
 	static from(params: CronJobParams) {
-		return new CronJob(
-			params.cronTime,
-			params.onTick,
-			params.onComplete,
-			params.start,
-			params.timeZone,
-			params.context,
-			params.runOnInit,
-			params.utcOffset,
-			params.unrefTimeout
-		);
+		// runtime check for JS users
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+		if (params.timeZone != null && params.utcOffset != null) {
+			throw new ExclusiveParametersError('timeZone', 'utcOffset');
+		}
+
+		if (params.timeZone != null) {
```

Comment:
I know this `if`/`else if`/`else` block is very verbose, but I couldn't find a solution to make TypeScript still understand the exclusive parameter thing when transitioning from the object parameters to the function parameters.
Exclusive parameters seem to work when providing them but do not restrict the type correctly inside the function using them.

Reply:
yeah I kinda want to deprecate the function constructor at some point. with as many options as we have keeping track of them is hard and leads to weird things like this code block. I'm fine holding off on that until v4 or something though

Reply:
stumbled upon `croner` docs today, the signature looks like `Cron(cronTime, onTick)` or `Cron(cronTime, options, onTick)`, I kinda like that